tests/static_reduction_map/reduction_functors_test.cu

```diff
@@ -0,0 +1,208 @@
+                 UsesExternalSync);
+}
+
+TEMPLATE_TEST_CASE_SIG(
+  "Test 'min' reduction functors",
+  "",
+  ((typename Value, typename Func, bool UsesExternalSync), Value, Func, UsesExternalSync),
+  (int32_t, cuco::detail::reduce_min_impl<int32_t>, false),
+  (int64_t, cuco::detail::reduce_min_impl<int64_t>, false),
+  (uint32_t, cuco::detail::reduce_min_impl<uint32_t>, false),
+  (uint64_t, cuco::detail::reduce_min_impl<uint64_t>, false),
+  (float, cuco::detail::reduce_min_impl<float>, true),
+  (double, cuco::detail::reduce_min_impl<double>, true),
+  (int32_t, thrust::minimum<int32_t>, true))
+{
+  test_case_impl(cuco::reduction_functor<Func, Value>(
+                   cuco::identity_value<Value>(std::numeric_limits<Value>::max())),
+                 thrust::minimum<Value>(),
+                 UsesExternalSync);
+}
+
+TEMPLATE_TEST_CASE_SIG(
+  "Test 'max' reduction functors",
+  "",
+  ((typename Value, typename Func, bool UsesExternalSync), Value, Func, UsesExternalSync),
+  (int32_t, cuco::detail::reduce_max_impl<int32_t>, false),
+  (int64_t, cuco::detail::reduce_max_impl<int64_t>, false),
+  (uint32_t, cuco::detail::reduce_max_impl<uint32_t>, false),
+  (uint64_t, cuco::detail::reduce_max_impl<uint64_t>, false),
+  (float, cuco::detail::reduce_max_impl<float>, true),
+  (double, cuco::detail::reduce_max_impl<double>, true),
+  (int32_t, thrust::maximum<int32_t>, true))
+{
+  test_case_impl(cuco::reduction_functor<Func, Value>(
+                   cuco::identity_value<Value>(std::numeric_limits<Value>::min())),
+                 thrust::maximum<Value>(),
+                 UsesExternalSync);
+}
+
+TEMPLATE_TEST_CASE_SIG(
+  "Test 'count' reduction functors",
+  "",
+  ((typename Value, typename Func, bool UsesExternalSync), Value, Func, UsesExternalSync),
+  (int32_t, cuco::detail::reduce_count_impl<int32_t>, false),
+  (int64_t, cuco::detail::reduce_count_impl<int64_t>, false),
+  (uint32_t, cuco::detail::reduce_count_impl<uint32_t>, false),
+  (uint64_t, cuco::detail::reduce_count_impl<uint64_t>, false))
+{
+  test_case_impl(cuco::reduction_functor<Func, Value>(cuco::identity_value<Value>(0)),
+                 equiv_count<Value>(),
+                 UsesExternalSync);
+}
+
+TEST_CASE("Test device lambda reduction functor", "")
+{
+  using Value   = int;
+  auto identity = cuco::identity_value<Value>(0);
+
+  auto lambda_add = [] __device__(Value const& lhs, Value const& rhs) noexcept {
+    return lhs + rhs;
+  };
+  test_case_impl(cuco::reduction_functor<decltype(lambda_add), Value>(identity, lambda_add),
+                 thrust::plus<Value>(),
+                 true);
+
+  using AtomicValue    = cuda::atomic<int, cuda::thread_scope_device>;
+  auto lambda_add_sync = [] __device__(AtomicValue & lhs, Value const& rhs) noexcept {
+    return lhs.fetch_add(rhs) + rhs;
+  };
+  test_case_impl(
+    cuco::reduction_functor<decltype(lambda_add_sync), Value>(identity, lambda_add_sync),
+    thrust::plus<Value>(),
+    false);
+}
```

Comment:
This test is still failing.

It seems that the signature of the device lambda does not match the expected signature of a synchronized functor:
https://github.com/NVIDIA/cuCollections/blob/ef1540178849a30ae8efd27b25d4f68a5bcab391/include/cuco/reduction_functors.cuh#L115-L143